contrib/nix/flake.nix

```diff
@@ -59,16 +64,49 @@
               # Darwin specific inputs
               pkgs.darwin.apple_sdk.frameworks.SystemConfiguration
             ];
+
+          src = fileSetForCrate ../..;
+          inherit version;
+        };
+
+        isClarityWASM = p: lib.hasPrefix "git+https://github.com/stacks-network/clarity-wasm.git" p.source;
+
+        cargoVendorDir = craneLib.vendorCargoDeps (
+          baseArgs
+          // {
+            # Use this function to override crates coming from git dependencies
+            overrideVendorGitCheckout =
+              ps: drv:
+              if lib.any (p: isClarityWASM p) ps then
+                drv.overrideAttrs
+                  (_old: {
+                    patches = [
+                      (builtins.fetchurl {
+                        url = "https://github.com/stacks-network/clarity-wasm/pull/627.patch";
+                        sha256 = "sha256:161mx1m21770lrsc2lfqlwzyydhy8f9bc7pmqb26rcki7s2ar31r";
+                      })
+                    ];
+                  })
+              else
+              # Nothing to change, leave the derivations as is
+                drv;
+
+            # Use this function to override crates coming from any registry checkout
+            overrideVendorCargoPackage = p: drv: drv;
+          }
+        );
+
+        commonArgs = baseArgs // {
+          inherit cargoVendorDir;
         };
 
         # Build *just* the cargo dependencies, so we can reuse
         # all of that work (e.g. via cachix) when running in CI
         cargoArtifacts = craneLib.buildDepsOnly (
           commonArgs
           // {
-            inherit version;
             pname = name;
-            src = fileSetForCrate ../..;
+            dummySrc = commonArgs.src;
```

Comment:
Can you add a comment to remember us why we need this? Effectively, this is disabling pre-building any cargo dependency. I think it's an unfortunate byproduct of how the workspace is structured right now, so not much we can do until we fix that.

Reply:
Unfortunately yes, because of how dependencies refer to each other (with WASM depending on `stacks-common`).